### PR TITLE
fix(curriculum): add missing tests for step 136 of JS RPG game 

### LINKED ISSUE #52780: Javascript Role Playing Game Bug on Step 136
### Describe the Issue

No matter what you write, you will succeed in this challenge, even if you choose to leave it blank.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-136

### Your code

```
function lose() {
// laksjflkjasdf
}
```

### Expected behavior

You can only pass if you do the following
```
function lose() {
  locations[5]
}
```

### Screenshots

_No response_

### System

- Device: PC
- OS: Windows 10 Pro
- Browser: Brave
- Version: 1.61.109 Chromium: 120.0.6099.144 64 bits

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pr2897**
Checklist:

- [x ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52780

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Hi @pr2897 ! 

Thank you for opening this PR.

I have updated the title of your post to something more descriptive so the maintainers know what this PR is about at first glance. 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md [hunk 1]
```diff
@@ -11,12 +11,19 @@ In the `lose` function, call the `update` function and pass in the sixth object
 
 # --hints--
 
-Test 1
+update method should be called
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md:14]:
```suggestion
You should call the `update` function inside the `lose` function.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md [hunk 2]
```diff
@@ -11,12 +11,19 @@ In the `lose` function, call the `update` function and pass in the sixth object
...
 
 ```js
+assert.match(lose.toString(), /update/)
+```
+
+locations[5] should be passed as argument to update method
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md:20]:
```suggestion
Your `update` function should have `locations[5]` for the argument.
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Add some comments to fix the language and missing back ticks for the hint text 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md [hunk 3]
```diff
@@ -11,12 +11,19 @@ In the `lose` function, call the `update` function and pass in the sixth object
 
 # --hints--
 
-Test 1
+You should call the `update` function inside the `lose` function.
 
 ```js
+assert.match(lose.toString(), /update/)
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md:17]: So technically this test only checks if the camper wrote `update` inside the `lose` function. 
It doesn't test if the camper calls it though.
We should update this test to check if the `update` function called

This should work

```suggestion
assert.match(lose.toString(), /update\(/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md [hunk 4]
```diff
@@ -11,12 +11,19 @@ In the `lose` function, call the `update` function and pass in the sixth object
...
+```
+
+Your `update` function should have `locations[5]` for the argument.
 
+```js
+assert.match(lose.toString(), /update\(locations\[5\]\)/)
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8f1d5f5ddbf74c07f733b.md:23]: add in missing semicolon 

```suggestion
assert.match(lose.toString(), /update\(locations\[5\]\)/);
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Last two small comments and then it looks good to me 👍

**Review by jdwilkin4 — APPROVED** [review]

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍

@pr2897 Thank you for the PR and congratulations on your first contribution to this repo 🎉
We look forward to your next.

Happy contributing!

**Pull request merged**